compiler/frontend/src/org/jetbrains/kotlin/resolve/lazy/descriptors/AbstractLazyMemberScope.kt

```diff
@@ -138,12 +139,12 @@ protected constructor(
         return propertyDescriptors(name)
     }
 
-    private fun doGetProperties(name: Name): Collection<PropertyDescriptor> {
+    private fun doGetProperties(name: Name): List<PropertyDescriptor> {
         val result = LinkedHashSet(declaredPropertyDescriptors(name))
 
         getNonDeclaredProperties(name, result)
 
-        return result.toList()
+        return sort(result.toList())
```

Comment:
Note: I don't think this should not incur measurable overhead. Sorting only when incremental compilation is enabled seems like an overkill.

Reply:
Ack. (Btw, I've changed where we sort this in my latest commit.)